### PR TITLE
Python 3 upgrade

### PR DESCRIPTION
**Pull request by Kazade**
Summary of changes proposed in this Pull Request:
- Update Djangae to Python 3 + Django 2.2
- Switch to running Djangae tests via tox
- Remove many many things:
  - djangae.db (moved to gcloud-connectors)
  - contrib.consistency (new datastore is strongly consistent)
  - contrib.mappers (partially superseded by djangae.tasks)
  - contrib.contenttypes (may come back, originally for eventual consistency reasons)
  - contrib.gauth* (will come back in some form, but there's no users API anymore)
  - contrib.processing (partially superseded by djangae.tasks)
  - contrib.uniquetool (unique markers don't exist as new datastore is consistent)
  - djangae.storage (will return, needs a rewrite)
  - djangae.mail (there's no built-in way to send email on Google Cloud)
  - djangae.test_runner / noseplugin (now we have separate emulators for cloud services)
  - djangae.fields (moved to gcloud-connectors)
  - djangae.forms (used for database fields which no longer exist in djangae)
  - lib.memcache (memcache doesn't exist on the Python 3 runtime)
- Write a new (optional) djangae.tasks app for working with Google Cloud Tasks

PR checklist:
- [x] Updated relevant documentation
- [x] Updated CHANGELOG.md 
- [x] Added tests for my change

### CODE REVIEW

**Comment by jacobg** [comment]
That's awesome! Amazing work @Kazade! Thank you!

Regarding the assumption of datastore consistency, would that require that Google have migrated an existing datastore to Firestore in Datastore mode? I haven't seen that Google has started that process yet. It's been marked by Google as "coming soon" already for well over a year.

**Comment by Kazade** [comment]
@jacobg Yes, all of this assumes Firestore in Datastore Mode, so it needs to be a new project, or the migration has to have happened.

I'm hoping they don't wait too much longer! I *really* don't want to have to reimplement all the stuff :)

**Comment by jacobg** [comment]
Thanks @Kazade! Also noticed that it's using a database-backed cache, with a note that memorystore is too slow. Are you able to elaborate?

https://github.com/potatolondon/djangae/blob/615aa411af16cc5ab21816d44d8ae708bcb842cf/djangae/settings_base.py#L2-L6

**Comment by Kazade** [comment]
So I only heard a rumour early-on that Memory Store access was slow from App Engine, but I don't know that for sure. Things might have changed.

I intend to ship some kind of Memory Store cache backend for Django, but possibly default to the database cache anyway to avoid people having to set-up yet another Google Cloud service to get started :)

**Comment by jacobg** [comment]
Thanks! I googled "memorystore slow", but didn't see anyone discuss that. Google does have a constraint of:

https://cloud.google.com/memorystore/docs/redis/product-constraints
> Cross-region connectivity
> You can connect only to a Cloud Memorystore for Redis instance from resources in the same region as your instance. For example, you can only connect to your Cloud Memorystore for Redis instance with a Compute Engine VM in the same region as your Cloud Memorystore for Redis instance.

I intend to replace my current use of App Engine Memcache with Memorystore.

File: README.md [hunk 1]
```diff
@@ -83,25 +79,12 @@ For pull request to be merged, following requirements should be met:
...
-
-Currently the default is 1.8. TravisCI runs on 1.8, 1.9, 1.10 and 1.11 currently.
-
-If you want to run the tests on a specific App Engine SDK version, then you can switch the installed version by doing:
-
-    $ SDK_VERSION=1.9.35 ./runtests.sh --install_sdk
-
-Note that this also re-installs the dependencies, so will reset the Django version to the default of 1.8.
-
+    $ tox -e py37
```
Comment (igniteflow) [inline README.md]: @Kazade I get the following exception running this locally

```
Traceback (most recent call last):
  File "/home/phil/Projects/potato/djangae/djangae/sandbox.py", line 46, in _wait
    response = urlopen("http://127.0.0.1:%s/" % port)
  File "/usr/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/usr/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.7/urllib/request.py", line 1345, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/usr/lib/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 111] Connection refused>
```

Do I need to spin up a local server?

Comment (igniteflow) [reply README.md]: Running inside a `Python 3.7.5rc1` virtualenv created with virtualenvwrapper `mkvirtualenv djangae-py3 --python=/usr/bin/python3`

Comment (interpegasus) [reply README.md]: **Line 46**
`response = urlopen("http://127.0.0.1:%s/" % port)`

Perhaps starting the app in a different server or port might help. Depending of the local configuration and the machine where the code is being executed. Hope it helps.
`[0.0.0.0 or localhost]:port`  might work.

**Pull request merged**